sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -0,0 +1,505 @@
+} az_iot_hub_client_options;
+
+/**
+ * @brief Azure IoT Hub Client.
+ *
+ */
+typedef struct az_iot_hub_client
+{
+  struct
+  {
+    az_span iot_hub_hostname;
+    az_span device_id;
+    az_iot_hub_client_options options;
+  } _internal;
+} az_iot_hub_client;
+
+/**
+ * @brief Gets the default Azure IoT Hub Client options.
+ * @details Call this to obtain an initialized #az_iot_hub_client_options structure that can be
+ *          afterwards modified and passed to #az_iot_client_init.
+ *
+ * @return #az_iot_hub_client_options.
+ */
+AZ_NODISCARD az_iot_hub_client_options az_iot_hub_client_options_default();
+
+/**
+ * @brief Initializes an Azure IoT Hub Client.
+ *
+ * @param[out] client
+ * @param[in] iot_hub_hostname The IoT Hub Hostname.
+ * @param[in] device_id The Device ID.
+ * @param[in] options A reference to an #az_iot_hub_client_options structure.  Can be NULL.
+ * @return #az_result.
+ */
+AZ_NODISCARD az_result az_iot_hub_client_init(
+    az_iot_hub_client const* client,
+    az_span iot_hub_hostname,
+    az_span device_id,
+    az_iot_hub_client_options const* options);
+
+/**
+ * @brief Gets the MQTT user name.
+ *
+ * @param[in] client
+ * @param[in] mqtt_user_name An empty #az_span with sufficient capacity to hold the MQTT user name.
+ * @param[out] out_mqtt_user_name The output #az_span containing the MQTT user name.
+ * @return #az_result.
+ */
+AZ_NODISCARD az_result az_iot_hub_client_user_name_get(
+    az_iot_hub_client const* client,
+    az_span mqtt_user_name,
+    az_span* out_mqtt_user_name);
+
+/**
+ * @brief Gets the MQTT client id.
+ *
+ * @param[in] client
+ * @param[in] mqtt_client_id An empty #az_span with sufficient capacity to hold the MQTT client id.
+ * @param[out] out_mqtt_client_id The output #az_span containing the MQTT client id.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_id_get(
+    az_iot_hub_client const* client,
+    az_span mqtt_client_id,
+    az_span* out_mqtt_client_id);
+
+/**
+ *
+ * SAS Token APIs
+ *
+ *   Use the following APIs when the Shared Access Key is available to the application or stored
+ *   within a Hardware Security Module. The APIs are not necessary if X509 Client Certificate
+ *   Authentication is used.
+ */
+
+/**
+ * @brief Gets the Shared Access clear-text signature.
+ * @details The application must obtain a valid clear-text signature using
+ *          this API, sign it using HMAC-SHA256 using the Shared Access Key as password then Base64
+ *          encode the result.
+ *
+ * @param[in] client
+ * @param[in] token_expiration_unix_time The time, in seconds, from the Unix epoch (1/1/1970).
+ * @param[in] signature An empty #az_span with sufficient capacity to hold the SAS signature.
+ * @param[out] out_signature The output #az_span containing the SAS signature.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_sas_signature_get(
+    az_iot_hub_client const* client,
+    uint32_t token_expiration_unix_time,

```

Comment:
>token_expiration_unix_time [](start = 13, length = 26)

Nit: epoch time instead of unix time?  I try to avoid hard-coding product/trademarky stuff when possible.